### DISCUSSION TITLE
Ctrl key press not properly transmitted to controlled end, affecting VSCode function jump

### DISCUSSION THREAD

**Discussion by mrjesen** [rustdesk/rustdesk#9011] [General]
### Bug Description

When using RustDesk for remote control, the "Ctrl key press" action does not seem to be properly transmitted to the controlled end. This issue is particularly noticeable when trying to use the function jump feature in Visual Studio Code, where holding down the Ctrl key and clicking on a function should trigger a jump to its definition.

### How to Reproduce

Establish a remote connection using RustDesk
Open Visual Studio Code on the controlled end
Try to use the function jump feature by holding down Ctrl and clicking on a function name

### Expected Behavior

The Ctrl key press should be recognized by the controlled end, allowing the function jump feature in VSCode to work as expected.

### Operating system(s) on local side and remote side

Windows 10

### RustDesk Version(s) on local side and remote side

1.2.3-2 -> 1.2.3-2

### Screenshots

No

**Comment by rustdesk** [comment]
@fufesou verify this. I tested ctrl + click to open context menu before, but ctrl + click to open function we may indeed miss, because we do not send modifier keys. Changing behavior this is too big for us, I choose hold on, look and see. Or add an option in the future for this, only send modifier keys if the option is enabled.

**Reply by mrjesen** [reply]
Apologies, I missed a detail. I am actually running VSCode through a virtual machine, which means the flow is: My PC -> RustDesk -> Win10 PC -> VMware -> Win10 Guest -> VSCode.

**Reply by mrjesen** [reply]
The combination of Ctrl + keyboard key works fine, but it seems that Ctrl + left mouse click is not being passed through.

**Reply by fufesou** [reply]
> My PC -> RustDesk -> Win10 PC -> VMware -> Win10 Guest -> VSCode.

This scenario has indeed not been tested. I'll test later.

**Reply by rustdesk** [reply]
No need to test this scenario.

**Comment by fufesou** [comment]
I've tested `1.3.0` -> `1.3.0` and `1.3.0` -> `1.2.3-2`, both win11 (but it should be the same to win10), Ctrl + left mouse button works fine.
I haven't tested `1.2.3-2` -> `1.2.3-2` because there's some strange connection bug.

https://github.com/user-attachments/assets/6cba4eb1-545d-4a23-9630-272ae0cf44de

**Comment by muwn** [comment]
This problem occurred again on 1.4.6. The function of Ctrl + left mouse button is no longer working.

**Reply by muwn** [reply]
Windows controls Mac, and both versions are 1.4.6.
